### PR TITLE
Slightly more understandable tokens() function

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Make a really confusing function a tiny bit less confusing!

Before
----------------------------------------
?!?!?!?

After
----------------------------------------
?!?!

Technical Details
----------------------------------------
Looked at "all" callers of this function and updated the variable names to match what is passed in.

Comments
----------------------------------------
@eileenmcnaughton

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
Eileen said : should & do are different - I have debugged code that does :-)

**Comment by eileenmcnaughton** [comment]
& then @demeritcowboy said `$b === 2` and @colemanw said 'but b is a cute variable name'

**Comment by mattwire** [comment]
Test fail looks a bit odd:
```
CRM_Export_BAO_ExportTest::testExportPhoneData
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'Much Much longer than just phone'
+'Phone'
```

**Comment by demeritcowboy** [comment]
Re: test fail: I wasn't sure if https://github.com/civicrm/civicrm-core/pull/31354 would really fix it. Obviously not.

Re: $b: For anyone who thinks Eileen has lost her mind, it's a reference to https://github.com/eileenmcnaughton/dompdf-fonts/pull/3/files

**Comment by eileenmcnaughton** [comment]
@demeritcowboy & is that to confirm that I've lost my mind for anyone who only thinks it?

**Comment by demeritcowboy** [comment]
Hehe maybe the comment should be restricted to just losing your mind about $b. Any other lost marbles are still possible.

**Comment by eileenmcnaughton** [comment]
Just noting that  [boo](https://github.com/civicrm/civicrm-core/commit/8189551c25e543e3a69f2ee19dbdd31edc68f1be#diff-f489283d43525602a5a90b8092c5189dd5d8b0392a30e789b4f99cd86eae1d69R103)  `!== $b`

**Comment by demeritcowboy** [comment]
`$marbleCount++`

File: Civi/Token/TokenRow.php [hunk 1]
```diff
@@ -144,31 +144,35 @@ public function context($a = NULL, $b = NULL) {
...
    * Don't worry - I've stared at it & all I see is a bunch of letters. However,
    * the answer to your problem is almost certainly that you are passing in null
    * rather than an empty string for 'c'.
+   * MJW said: I've renamed the bunch of letters so they might be a bit more meaningful.
+   * Also, I don't think this function should ever be called without $tokenField being set
+   * but tests do call it like that.
+   *
+   * @param string|array $tokenEntity
+   * @param string|array $tokenField
+   * @param string|array $tokenValue
```
Comment (dontub) [inline Civi/Token/TokenRow.php:157]: Actually `$tokenValue` can also be `\DateTime` and `\Brick\Money\Money`. Those objects are handled in [TokenProcessor](https://github.com/civicrm/civicrm-core/blob/161e790ebd2c41ca651db8b2df8c335d4fe2264f/Civi/Token/TokenProcessor.php#L480).

BTW: I'm wondering what kind of arrays are allowed and what the expected result is.

Comment (dontub) [reply Civi/Token/TokenRow.php:157]: PR to fix this: #31584.

**Pull request merged**